apps/api/src/coscrad-cli/seed-test-data-with-command.cli-command.ts

```diff
@@ -0,0 +1,147 @@
+    payloadOverrides?: Record<string, unknown>;
+}
+
+/**
+ * TODO Move this to the test data directory. We may want to inject a `TestDataManager`
+ * service that can find these for us.
+ */
+const allFsas = [
+    {
+        type: 'CREATE_SONG',
+        payload: {
+            aggregateCompositeIdentifier: { id: 'dummy-song', type: AggregateType.song },
+            title: 'test-song-name (language)',
+            titleEnglish: 'test-song-name (English)',
+            lyrics: 'la la la',
+            audioURL: 'https://www.mysound.org/song.mp3',
+        },
+    },
+];
+
+@CliCommand({
+    name: 'seed-test-data-with-command',
+    description: 'executes a fixture command (by type) with optional overrides to seed test data',
+})
+export class SeedTestDataWithCommand extends CliCommandRunner {
+    constructor(
+        private readonly commandHandlerService: CommandHandlerService,
+        @Inject(ID_MANAGER_TOKEN) private readonly idManager: IIdManager,
+        @Inject(COSCRAD_LOGGER_TOKEN) private readonly logger: ICoscradLogger
+    ) {
+        super();
+    }
+
+    async run(_passedParams: string[], options: SeedTestDataWithCommandOptions): Promise<void> {
+        const { type: commandType, payloadOverrides } = options;
+
+        const fixtureFsaSearchResult = allFsas.find(({ type }) => type === commandType);
+
+        if (isUndefined(fixtureFsaSearchResult)) {
+            this.logger.log(`failed to find a fixture command of type: ${commandType}. Exiting.`);
+
+            throw new InternalError(`No fixture command of type: ${commandType} found`);
+        }
+
+        const { payload: defaultPayload } = fixtureFsaSearchResult;
+
+        const fsaToWithOverrides = {
+            type: commandType,
+            payload: {
+                ...defaultPayload,
+                ...(payloadOverrides || {}),
+            },
+        };
+
+        this.logger.log(`attempting to execute FSA: ${JSON.stringify(fsaToWithOverrides)}`);
+
+        /**
+         * TODO We need to do this conditionally, only for create commands. For
+         * this reason, it is better if this logic gets moved to a special method
+         * on the command handler or a similar service. We don't want this logic
+         * in the CLI command controller in case we need to expose it via some
+         * other transport layer.
+         */
+        const generatedId = await this.idManager.generate();
+
+        const { payload: payloadWithoutGeneratedId } = fsaToWithOverrides;
+
+        const fsaToExecute = {
+            ...fsaToWithOverrides,
+            payload: {
+                ...payloadWithoutGeneratedId,
+                aggregateCompositeIdentifier: {
+                    ...payloadWithoutGeneratedId.aggregateCompositeIdentifier,
+                    id: generatedId,
+                },
+            },
+        };
+
+        const result = await this.commandHandlerService.execute(fsaToExecute, {
+            // TODO Assign the following to a constant
```

Comment:
Do this now.

Reply:
Note that this could be behaviour on a test data manager service method that isn't actually available in the regular path. We probably don't want this logic here in the cli command.